src/crimson/os/seastore/transaction_manager.h

```diff
@@ -644,6 +644,93 @@ class TransactionManager : public ExtentCallbackInterface {
     ExtentPlacementManager::dispatch_result_t dispatch_result,
     std::optional<journal_seq_t> seq_to_trim = std::nullopt);
 
+  /**
+   * alloc_existing_extent
+   *
+   * Allocates a new extent at given existing_paddr that must be absolute and 
+   * use the buffer to fill the new extent if buffer exists. Otherwise, will
+   * not read disk to fill the new extent.
+   * Returns the new extent.
+   */
+  using alloc_existing_extent_iertr =
+    alloc_extent_iertr::extend_ertr<Device::read_ertr>;
+  template <typename T>
+  using alloc_existing_extent_ret =
+    alloc_existing_extent_iertr::future<TCachedExtentRef<T>>;
+  template <typename T>
+  alloc_existing_extent_ret<T> alloc_existing_extent(
+    Transaction &t,
+    laddr_t existing_laddr,
+    paddr_t existing_paddr,
+    extent_len_t length,
+    laddr_t laddr,
```

Comment:
Use the unnambiguous names: original_laddr, original_paddr, remap_length, remap_laddr.